### ISSUE TITLE
The results of Flow4d and DeFlow on the validation set do not match those in the Flow4d paper.

### ISSUE THREAD

**Issue by ZhonghuaYi** [KTH-RPL/OpenSceneFlow#9]
Thank you for your contributions to this project. And If this issue should not be raised in this repository but should be directed to the official repository of flow4d, please let me know.

I'm currently trying to reproduce the results in the Flow4d paper. Since the test split is unknown, I can only validate the models on the val split.
My enviroment is build upon docker which is provided from this repo, and the code works fine.

However, after I run the validation script to validate the Flow4d checkpoint and DeFlow checkpoint (all download from this repo), I get the following result:

| Validate metric          | Flow4d                   | Deflow                   |
|--------------------------|--------------------------|--------------------------|
| val/Dynamic/BACKGROUND  | nan                      | nan                      |
| val/Dynamic/CAR         | 0.14861367441300755      | 0.15406067201746712      |
| val/Dynamic/Mean        | 0.1944707564173708       | 0.2503908338018606       |
| val/Dynamic/OTHER_VEHICLES | 0.16510866209490632      | 0.16919413593442598      |
| val/Dynamic/PEDESTRIAN   | 0.2282272031203494       | 0.38746830627256534      |
| val/Dynamic/WHEELED_VRU  | 0.23593348604122         | 0.29084022098298395      |
| val/EPE_BS               | 0.008650692479973474     | 0.007429371694968465     |
| val/EPE_FD               | 0.0864265558392112       | 0.08880582888497979      |
| val/EPE_FS               | 0.016565995754708962     | 0.01942031611276835      |
| val/IoU                  | 0.6577074701800952       | 0.6565594548930822       |
| val/Static/BACKGROUND   | 0.008516064793231169     | 0.007196461429879709     |
| val/Static/CAR          | 0.013617087483902685     | 0.015358700074269557     |
| val/Static/Mean         | 0.014316466542977135     | 0.0164002733463524       |
| val/Static/OTHER_VEHICLES | 0.01819301479489148      | 0.021960908274204893     |
| val/Static/PEDESTRIAN    | 0.01732545062823507      | 0.02297448210230759      |
| val/Static/WHEELED_VRU   | 0.013930715014625275     | 0.014510814851100251     |
| val/Three-way            | 0.03721441469129788      | 0.038551838897572195     |

However, the validation result is also provided in the Flow4d paper, which is shown below:

![Image](https://github.com/user-attachments/assets/8e8d3df3-92c1-4602-8721-a35e2080cf4b)

In addition, the validation result of these two models are also provided in the MambaFlow paper, which is as following:

![Image](https://github.com/user-attachments/assets/9ec61087-9adf-4409-8742-fbcb1e0e38f9)

The result obtained by me seems too far from those from these two papers. Flow4d performs generally worse in my implementation, while DeFlow is on the contrary. But Flow4d still performs better than DeFlow with a slight leading advantage.

By the way, I retrained Flow4d using the training command provided by this repo, and the result is similar to the downloaded checkpoint.

Now I wonder whether this result is acceptable due to the device difference. If not, which part may be the potential issuse that leads to this strange result?

Thank you!

**Comment by Kin-Zhang** [comment]
I think you may need check this pull request: https://github.com/KTH-RPL/OpenSceneFlow/pull/5

**Comment by Kin-Zhang** [comment]
Especially this comment in the pull request: https://github.com/KTH-RPL/OpenSceneFlow/pull/5#issuecomment-2709845736

**Comment by ZhonghuaYi** [comment]
Thank you! So my result is right after using the fixed ground truth generation method. This really solve my question.

**Comment by Kin-Zhang** [comment]
I will also update the pull request table in our HiMo paper soon (Appendix, I think). So you can link to the pull request/cite paper to indicate why scores are different etc. Let me know if there is any issue still.

**Comment by ZhonghuaYi** [comment]
Sure, this will be a good reference. Thanks for your help :).
